Mathlib/Data/Finset/Sum.lean

```diff
@@ -94,4 +96,83 @@ theorem disj_sum_strictMono_right (s : Finset α) :
     StrictMono (s.disjSum : Finset β → Finset (α ⊕ β)) := fun _ _ =>
   disjSum_ssubset_disjSum_of_subset_of_ssubset Subset.rfl
 
+lemma disjSum_inj {α β : Type*} {s₁ s₂ : Finset α} {t₁ t₂ : Finset β} :
+    s₁.disjSum t₁ = s₂.disjSum t₂ ↔ s₁ = s₂ ∧ t₁ = t₂ := by
+  simp [Finset.ext_iff]
+
+lemma Injective2_disjSum {α β : Type*} : Function.Injective2 (@disjSum α β) :=
+  fun _ _ _ _ => by simp [Finset.ext_iff]
+
+/--
+Given a finset of elements `α ⊕ β`, extract all the elements of the form `α`. This
+forms a quasi-inverse to `disjSum`, in that it recovers its left input.
+-/
+def toLeft (s : Finset (α ⊕ β)) : Finset α :=
```

Comment:
Do we have the `Option` version of this? If so, a docstring reference would be good.

Reply:
Loogle can't find it, so I don't think so

Reply:
`List.reduceOption` is what I'm thinking of.

Reply:
There's also List.partitionMap which is a closer analogy, but it has basically no API, in particular not nearly enough to show you get something well-defined on multisets

Reply:
Nice find. Can you leave a note somewhere in this file about that function? No need to write any new theorems in this PR, but I think `s.toLeft` is morally `(partitionMap id s).1`?

Reply:
Morally it is - indeed my first attempt at this definition was in terms of that, but it's way more painful to use in context. Will add a comment